server/src/main/java/org/spine3/server/event/EventStorage.java

```diff
@@ -57,11 +66,31 @@ public Event apply(@Nullable EventEntity input) {
                 }
             };
 
+    @Override
+    protected org.spine3.server.storage.EventStorage createStorage(StorageFactory factory) {
+        final RecordStorage<EventId> recordStorage = super.createStorage(factory);
+        final org.spine3.server.storage.EventStorage storage =
+                factory.createEventStorage(recordStorage);
+        return storage;
+    }
+
+    @Nonnull
+    @Override
+    protected org.spine3.server.storage.EventStorage recordStorage() {
+        return (org.spine3.server.storage.EventStorage) super.recordStorage();
+    }
+
     Iterator<Event> iterator(EventStreamQuery query) {
-        final Iterator<EventEntity> filtered = iterator(createEntityFilter(query));
-        final List<EventEntity> entities = Lists.newArrayList(filtered);
-        Collections.sort(entities, EventEntity.comparator());
-        final Iterator<Event> result = Iterators.transform(entities.iterator(), getEventFunc());
+        final org.spine3.server.storage.EventStorage storage = recordStorage();
+        final Map<EventId, EntityRecord> records = storage.readAll(query);
+        final Collection<EventEntity> entities = transform(records.entrySet(),
+                                                           storageRecordToEntity());
+        // TODO:2017-05-19:dmytro.dashenkov: Remove after the Entity Column approach is implemented.
```

Comment:
@dmdashenkov please create an issue in `core-java` right after this PR is merged.